### PR TITLE
Register nntile SDPA via F.scaled_dot_product_attention

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

- Register `device="nntile"` support for `F.scaled_dot_product_attention` through PyTorch's PrivateUse1 overrideable SDPA path (`_fused_sdp_choice`, `_scaled_dot_product_fused_attention_overrideable`, backward).
- ATen forward/backward delegate to the existing `sdpa_forward` / `sdpa_backward` kernel (maxsumexp + softmax_inplace internally). Forward returns a placeholder `logsumexp` tensor (OpenReg API requirement only); backward **ignores** it and uses internal **maxsumexp** buffers via `sdpa_backward` — not logsumexp.
- Refactor `torch_nntile.nn.sdpa_eager` / `SDPA` to: cyclic transpose → `F.scaled_dot_product_attention` → cyclic transpose (replacing direct `_NntileSdpaEager` pybind path in Python).

## Architecture

```
sdpa_eager (projection layout B,S,D,H)
  → nntile_model_transpose(1) → (H,B,S,D)
  → F.scaled_dot_product_attention  [ATen overrideable → sdpa_forward]
  → nntile_model_transpose(3) → (B,S,D,H)
```

HF-style callers can also call `F.scaled_dot_product_attention` directly on `(batch, n_heads, seq, head_size)` tensors on `device="nntile"`.

## v1 limits (unchanged)

- float32 only
- no dropout, GQA, or custom scale
- `is_causal` and bool/float `attn_mask` supported (converted to bool `[q_seq, k_seq]`)

## Tests

- New `torch_nntile/tests/test_sdpa_aten.py`: direct `F.sdpa` forward/backward, causal/bool masks, rejection tests, sdpa_eager integration
- Existing `test_sdpa_parity.py`: all 11 tests pass (now exercises transpose → F.sdpa path)

```
pytest -vv torch_nntile/tests/test_sdpa_aten.py torch_nntile/tests/test_sdpa_parity.py
# 20 passed
```

<div><a href="https://cursor.com/agents/bc-c8ccf499-d31e-4222-9f6e-fc0ff21c019e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-c8ccf499-d31e-4222-9f6e-fc0ff21c019e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review

**Comment by Muxas** [comment]
@cursor review
